go/workflow.go

```diff
@@ -0,0 +1,333 @@
+package loyalty
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+)
+
+const TaskQueue = "CustomerLoyaltyTaskQueue"
+const EventsThreshold = 10000
+
+// Signal, query, and error string constants
+const (
+	SignalCancelAccount       = "cancelAccount"
+	SignalAddPoints           = "addLoyaltyPoints"
+	SignalInviteGuest         = "inviteGuest"
+	SignalEnsureMinimumStatus = "ensureMinimumStatus"
+	QueryGetStatus            = "getStatus"
+	QueryGetGuests            = "getGuests"
+)
+
+const (
+	EmailWelcome            = "Welcome to our loyalty program! You're starting out at '%v' status."
+	EmailGuestCanceled      = "Sorry, your guest has already canceled their account."
+	EmailGuestInvited       = "Congratulations! Your guest has been invited!"
+	EmailInsufficientPoints = "Sorry, you need to earn more points to invite more guests!"
+	EmailPromoted           = "Congratulations! You've been promoted to '%v' status!"
+	EmailDemoted            = "Unfortunately, you've lost enough points to bump you down to '%v' status. 😞"
+	EmailCancelAccount      = "Sorry to see you go!"
+)
+
+type GuestAlreadyCanceledError struct {
+	msg string
+}
+
+func (e *GuestAlreadyCanceledError) Error() string { return e.msg }
+
+func CustomerLoyaltyWorkflow(ctx workflow.Context, customer CustomerInfo) (err error) {
```

Comment:
```suggestion
func CustomerLoyaltyWorkflow(ctx workflow.Context, customer CustomerInfo) error {
```

No need to name the return type